assets/js/delete-task.js

```diff
@@ -0,0 +1,9 @@
+import { CLICK_EVENT } from "/assets/js/constants.js";
+
+function deleteTaskEvent(deleteButton, task, taskList, allTasks, id) {
+  deleteButton.addEventListener(CLICK_EVENT, () => {
```

Comment:
the event listener is not unregistered from the DOM hence causing memory leacks. It is also very late to register the event handler upon the delete button click.

```
// task delete handler should be registered upon the creation of the task
deleteButton.addEventListener(CLICK_EVENT, deleteTask)

function deleteTask() {
    taskList.removeChild(task);
    delete allTasks[id];
    
    // cleanup the listener when the associated DOM node for the task element is no longer available in memory
    deleteButton.removeEventListener(CLICK_EVENT, deleteTask)
}

```

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener